library/src/main/java/com/novoda/simplechromecustomtabs/SimpleChromeCustomTabs.java

```diff
@@ -20,42 +19,35 @@
 
 public final class SimpleChromeCustomTabs implements WebNavigator, Connection, AvailableAppProvider {
 
-    private static Context applicationContext;
     private Connection connection;
     private WebNavigator webNavigator;
     private AvailableAppProvider availableAppProvider;
 
-    private SimpleChromeCustomTabs() {
-        //no-op
-    }
-
-    private static class LazyHolder {
-        private static final SimpleChromeCustomTabs INSTANCE = new SimpleChromeCustomTabs();
+    /**
+     * @deprecated There is no need to call it anymore in order to use this library,
+     * use {@link #getInstance()} instead
+     */
+    @Deprecated public static void initialize(Context context) {
+        // do nothing
     }
 
     public static SimpleChromeCustomTabs getInstance() {
-        if (applicationContext == null) {
-            throw new DeveloperError("SimpleChromeCustomTabs must be initialized. Use SimpleChromeCustomTabs.initialize(context)");
-        }
-
-        return LazyHolder.INSTANCE;
+        return LazyHolder.INSTANCE.getSimpleChromeCustomTabs();
     }
 
-    public static void initialize(Context context) {
-        applicationContext = context.getApplicationContext();
-        LazyHolder.INSTANCE.connection = Connection.Creator.create();
-        LazyHolder.INSTANCE.webNavigator = WebNavigator.Creator.create();
-        LazyHolder.INSTANCE.availableAppProvider = AvailableAppProvider.Creator.create();
+    private static class LazyHolder {
+        private static final SimpleChromeCustomTabsProvider INSTANCE = new SimpleChromeCustomTabsProvider();
     }
 
-    public Context getContext() {
-        return applicationContext;
+    void injectModules(Connection connection, WebNavigator webNavigator, AvailableAppProvider availableAppProvider) {
+        this.connection = connection;
+        this.webNavigator = webNavigator;
+        this.availableAppProvider = availableAppProvider;
     }
 
     /**
      * Provides a {@link NavigationFallback} to specify navigation mechanism in case of no Chrome Custom Tabs support found.
      *
-     * @param navigationFallback
```

Comment:
A navigation fallback is basically a callback that gets executed when the navigation using chrome custom tabs is not possible for any reason (the device doesn't have a compatible version of chrome installed or any other reason)